### DISCUSSION TITLE
PAT token in .npmrc

### DISCUSSION THREAD

**Discussion by kirtanchandak** [sugarlabs/musicblocks-v4#241] [Q&A] [answered]
Hey, I added my PAT token in .npmrc file, but as soon as I run "npm ci" I am getting an error. 
![Screenshot 2023-01-09 154255](https://user-images.githubusercontent.com/92196705/211285043-1e6b7212-048e-45bd-a669-501e93a11713.png)

**Comment by meganindya** [comment]
Try putting
```shell
//npm.pkg.github.com/:_authToken=<YOUR_GITHUB_AUTH_TOKEN>
```
in your `~/.npmrc` file

I'm not sure what adding an `echo` in the `.npmrc` in the repo does

On Windows, the `.npmrc` file will most likely be at `%USERPROFILE%\.npmrc`

**Reply by Tushar98644** [reply]
> Try putting
> 
> ```shell
> //npm.pkg.github.com/:_authToken=<YOUR_GITHUB_AUTH_TOKEN>
> ```
> 
> in your `~/.npmrc` file
> 
> I'm not sure what adding an `echo` in the `.npmrc` in the repo does
> 
> On Windows, the `.npmrc` file will most likely be at `%USERPROFILE%\.npmrc`

even after doing that there's still the authentication issue 
<img width="1440" alt="Screenshot 2023-01-17 at 8 22 18 PM" src="https://user-images.githubusercontent.com/107763774/212931235-32eefd5a-3b36-4d27-b4bb-aa462f42d6ac.png">

**Reply by likitarai1** [reply]
Hi @Tushar98644 Were you able to solve this error?

**Reply by Tushar98644** [reply]
yes ....the mistake i was doing on my part was putting the pat token inside `[]`....if you are doing the same, just remove them and hopefully the error will be resolved ...

**Comment by Anurag-Band** [comment]
Hii @kirtanchandak, Is this solution worked for you ??

platform: Ubuntu 22.04 .1 LTS

When I was setting the local environment using Docker I also got similar error, and the above stated solution by @meganindya is not worked for me ??
![Screenshot from 2023-01-12 11-07-25](https://user-images.githubusercontent.com/95676076/211986466-041cb995-2f70-4e45-8a47-56672e032f11.png)

& The versions that I getting from building the Compose Image form Docker File is:

![Screenshot from 2023-01-12 11-11-26](https://user-images.githubusercontent.com/95676076/211986797-fcba32a5-1618-4a72-b6ac-9afd8782b251.png)

Not getting which is mentioned in the Development documentation 

Can you please give me Information about the current versions that Musicblocks V4 is using ??

**Comment by meganindya** [comment, marked as answer]
I can see

```shell
npm ERR! 401 Unauthorized - GET https: //npm.pkg.github.com/dow.... - authentication token not provided
```

So, it seems you don't have your PAT in the `~/.npmrc` file in your local machine or inside the docker container.

I tried to replicate it.

![Screenshot 2023-01-12 at 3 30 29 PM](https://user-images.githubusercontent.com/33328728/212038256-c9c90073-129a-425c-999c-e0132014f7e2.png)

In the first attempt, I removed the PAT from my local machine's `~/.npmrc` and as expected, it throws a 401 error.

In the second attempt, I added the PAT in my docker container's `~/.npmrc` and it succeeds.

I found that you don't necessarily need to add your PAT in the docker container's `~/.npmrc` if you have that in your local machine's `~/.npmrc`. However, if it doesn't work for you, add it after opening your docker container.

```shell
echo "//npm.pkg.github.com/:_authToken=<YOUR_PAT>" >> ~/.npmrc
```

**Reply by Anurag-Band** [reply]
Hiii @meganindya, Thanks for helping me out your Answer is Helpful

But, I think the actual problem is with npm registry login,

The Solution is can be found in the below Stackoverflow Answer that helped me to resolve it :-

[https://stackoverflow.com/questions/70104429/npm-install-error-401-when-installing-a-private-repository](url)

**Reply by meganindya** [reply]
It's not a private repository 😛

Are you able to setup?

**Reply by Anurag-Band** [reply]
yeah I know @meganindya, 

But somehow it worked 🙃 !!, so I shared for others who will get same problem 

I have setup the local environment, looking forward to Contribute to the Project, & we will build crazy features 😇
